### ISSUE TITLE
Warning when Search Terms page is opened by clicking option  at the footer

### ISSUE THREAD

**Issue by ghost** [magento/magento2#25245]
### Preconditions (*)
1. Fresh Magento installation with magento 2.3-develop

### Steps to reproduce (*)
1. Setup vanilla 2.3-develop
2. Create a product in the backend, give it the name "test"
3. On the frontend, perform a search query, search for "test"
4. Notice that it finds the product
5. If you now go to the Search Terms page in the frontend, it will show the search term "test" => **good**
6. Now in the backend, go to Marketing > SEO & Search > Search Terms
7. Open your "test" term
8. Set the "Number of Uses" to 0 and save
9. Click On Search terms option present at the bottom

### Expected result (*)
1. Page opens without any error
2. There is no errors in the system log 

### Actual result (*)
1. In developer Magento mode  Warning is shown on the Search Terms page
`Warning: natcasesort() expects parameter 1 to be array, null given in /var/www/html/deployer/instances/i-25245-2-3-develop/app/code/Magento/Search/Block/Term.php on line 102 [] []`
2. In default mode error is shown in `system.log`

### Additional information
Based on comment https://github.com/magento/magento2/issues/25245#issuecomment-546703776
Fix seems to be very easy:
```diff
diff --git a/app/code/Magento/Search/Block/Term.php b/app/code/Magento/Search/Block/Term.php
index b27ef6b01fd..a1eb0bb64c9 100644
--- a/app/code/Magento/Search/Block/Term.php
+++ b/app/code/Magento/Search/Block/Term.php
@@ -87,6 +87,7 @@ class Term extends Template
                 return $this;
             }

+            $termKeys = [];
             $this->_maxPopularity = reset($terms)->getPopularity();
             $this->_minPopularity = end($terms)->getPopularity();
             $range = $this->_maxPopularity - $this->_minPopularity;
```

**Comment by ghost** [comment]
@magento give me 2.3-develop instance

**Comment by krishprakash** [comment]
@magento give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @krishprakash. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @krishprakash, here is your [Magento instance](https://i-25245-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-25245-2-3-develop.instances.magento-community.engineering/admin](https://i-25245-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by krishprakash** [comment]
Hi @vishalverma279. Thank you for your report.
Not able to reproduce this issue as provided steps by you.
Please provide more details and confirm that issue is reproduced on venilla magento instance.
Thanks

**Comment by ghost** [comment]
@krishprakash possibly warnigs are disabled at the docker instance of magento. can you please check by cloning the same at your locals server
https://drive.google.com/file/d/1rysG1boY1TobKhajeuHoBK5IgoL47zXL/view

**Comment by krishprakash** [comment]
@vishalverma279. checked at local 2.3-develop and 2.3.3  instances but could not reproduce this issue.
https://www.screencast.com/t/MHbrrq3Chd

**Comment by krishprakash** [comment]
Hi @vishalverma279 feel free to reopen the if you confirm that issue is reproduced on venilla magento instance.
Thanks

**Comment by hostep** [comment]
I'm reopening this, here are the steps to reproduce:

1. Setup vanilla 2.3-develop
2. Create a product in the backend, give it the name "test"
3. On the frontend, perform a search query, search for "test"
4. Notice that it finds the product
5. If you now go to the Search Terms page in the frontend, it will show the search term "test" => **good**
6. Now in the backend, go to Marketing > SEO & Search > Search Terms
7. Open your "test" term
8. Set the "Number of Uses" to 0 and save
9. If you now go to the Search Terms page in the frontend, you'll see the error `Exception #0 (Exception): Warning: natcasesort() expects parameter 1 to be array, null given in app/code/Magento/Search/Block/Term.php on line 102` => **not good**

Fix seems to be very easy:
```diff
diff --git a/app/code/Magento/Search/Block/Term.php b/app/code/Magento/Search/Block/Term.php
index b27ef6b01fd..a1eb0bb64c9 100644
--- a/app/code/Magento/Search/Block/Term.php
+++ b/app/code/Magento/Search/Block/Term.php
@@ -87,6 +87,7 @@ class Term extends Template
                 return $this;
             }

+            $termKeys = [];
             $this->_maxPopularity = reset($terms)->getPopularity();
             $this->_minPopularity = end($terms)->getPopularity();
             $range = $this->_maxPopularity - $this->_minPopularity;
```

If somebody wants to create a PR with this fix, that would be great! 🙂

**Comment by sdzhepa** [comment]
@magento give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @sdzhepa. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @sdzhepa, here is your [Magento instance](https://i-25245-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-25245-2-3-develop.instances.magento-community.engineering/admin](https://i-25245-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by sdzhepa** [comment]
The issue can be reproduced by steps from https://github.com/magento/magento2/issues/25245#issuecomment-546703776

`Warning: natcasesort() expects parameter 1 to be array, null given in /var/www/html/deployer/instances/i-25245-2-3-develop/app/code/Magento/Search/Block/Term.php on line 102 [] []`

Is shown on the Search Terms page in Developer mode, in default mode, such error can be found in `system.log`

I updated the Issue description and confirmed.
Thank you @hostep for detailed steps to reproduce and root cause investigation!

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @sdzhepa 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-22170` were created

**Issue Available**: @sdzhepa, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by alzhr** [comment]
Hi

**Comment by 24rtj** [comment]
I’m

**Comment by magento-engcom-team** [comment]
Hi @vishalverma279. Thank you for your report. 
The issue has been fixed in magento/magento2#25246 by @vishalverma279 in 2.4-develop branch
Related commit(s):
 - [6b3bd36cb4fda9a9f162f123932c6b9f93291ae9](https://github.com/magento/magento2/commit/6b3bd36cb4fda9a9f162f123932c6b9f93291ae9)
 - [c7483ba1085ed271e0e098de55bc1092a4910f09](https://github.com/magento/magento2/commit/c7483ba1085ed271e0e098de55bc1092a4910f09)
 - [7b99e510a4feaa5fdde059e4e87668a43e3d348c](https://github.com/magento/magento2/commit/7b99e510a4feaa5fdde059e4e87668a43e3d348c)
 - [c58a7e3b74e61064988326a344252f52eb4c11bb](https://github.com/magento/magento2/commit/c58a7e3b74e61064988326a344252f52eb4c11bb)
 - [1e14f000e8b32a6a0a1101879b5632af1f733cfb](https://github.com/magento/magento2/commit/1e14f000e8b32a6a0a1101879b5632af1f733cfb)
 - [17600435ec7aed34dc93fcbda71f3bf8a2e1db09](https://github.com/magento/magento2/commit/17600435ec7aed34dc93fcbda71f3bf8a2e1db09)
 - [d1f1ebaf34dab66e16acf52a4a0b135282399d1c](https://github.com/magento/magento2/commit/d1f1ebaf34dab66e16acf52a4a0b135282399d1c)

The fix will be available with the upcoming 2.4.0 release.
